joeynmt/transformer_layers.py

```diff
@@ -200,8 +205,13 @@ def forward(self, x: Tensor, mask: Tensor, x_p: Tensor, p_mask: Tensor) -> Tenso
         x_norm = self.layer_norm(x)
         x_p_norm = self.layer_norm(x_p)
         h = self.src_src_att(x_norm, x_norm, x_norm, mask)
-        h2 = self.src_src_att(x_p_norm, x_p_norm, x_p_norm, p_mask)
-        h = self.gated_sum(self.dropout(h) + x, self.dropout(h2) + x_p)
+        h2 = self.src2_src_att(x_p_norm, x_p_norm, x_norm, p_mask)
+        #print(x.size(), x_p.size()) #torch.Size([13, 15, 64]) torch.Size([13, 34, 64])
+        #print(h.size(), h2.size()) #torch.Size([13, 15, 64]) torch.Size([13, 15, 64])
+        #h = self.gated_sum(self.dropout(h) + x, self.dropout(h2) + x_p)
+        #print(torch.cat((h, h2), -1).size()) #torch.Size([9, 27, 128])
+        g = torch.sigmoid(self.W_g( torch.cat((h, h2), -1) ) + self.b_g)
```

Comment:
If we're doing L -> R order, might want to switch arguments of h and h2
```suggestion
        g = torch.sigmoid(self.W_g( torch.cat((h2, h), -1) ) + self.b_g)
```